File: ros2/paquetes/ej1/tails_threads/tails_threads_py/tails_threads_py/subscriber.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String
import time

class SimpleSubscriber(Node):
    def __init__(self):
        super().__init__('simple_subscriber')
        self.subscription = self.create_subscription(
            String,
            'chatter',
            self.callback,
            100  # Ajusta la profundidad de la cola aquí
        )

    def callback(self, msg):
        self.get_logger().info(f'Received: {msg.data}')
        time.sleep(1)  # Simula un procesamiento lento

def main(args=None):
    rclpy.init(args=args)
    node = SimpleSubscriber()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()

File: ros2/paquetes/ej1/tails_threads/tails_threads_cpp/src/subscriber.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"
#include <chrono>
#include <thread>

class SimpleSubscriber : public rclcpp::Node {
public:
    SimpleSubscriber() : Node("simple_subscriber") {
        subscription_ = this->create_subscription<std_msgs::msg::String>(
            "chatter",
            10,  // Ajusta la profundidad de la cola aquí
            [this](const std_msgs::msg::String::SharedPtr msg) {
                RCLCPP_INFO(this->get_logger(), "Received: '%s'", msg->data.c_str());
                std::this_thread::sleep_for(std::chrono::seconds(1));  // Simula un procesamiento lento
            });
    }

private:
    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription_;
};

int main(int argc, char *argv[]) {
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<SimpleSubscriber>());
    rclcpp::shutdown();
    return 0;
}
